modules/rate-limiting/keeper/rate_limit.go

```diff
@@ -83,7 +84,11 @@ func (k Keeper) AddRateLimit(ctx sdk.Context, msg *types.MsgAddRateLimit) error
 	// Confirm the channel exists
 	_, found = k.channelKeeper.GetChannel(ctx, transfertypes.PortID, msg.ChannelId)
 	if !found {
-		return types.ErrChannelNotFound
+		// Check if the channelId is clientId
+		status := k.clientKeeper.GetClientStatus(ctx, msg.ChannelId)
+		if status == ibcexported.Unauthorized {
+			return types.ErrChannelNotFound
```

Comment:
This is super hacky. The field is called `ChannelId`. Should we not add a new protobuf field for the client ID and make them mutually exclusive?

Reply:
This would probably make msg validation easier too. There is a function to validate client ID, so you can validate the mutual exclusiveness + whichever is preset you either validate as channel or client id

Reply:
added back the tests. renamed external facing channelId to channelOrClientId